controllers/auth_config_controller.go

```diff
@@ -93,6 +93,9 @@ func (r *AuthConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 			return ctrl.Result{}, err
 		}
 
+		// delete related authconfigs from cache.
+		r.Cache.Delete(cacheId)
```

Comment:
This is the simplest solution.

Another approach would be to implement an alternate delete function that takes as input the cache key (host) instead of the cache id. We could then delete only the diff between the keys of the `evaluatorConfigByHost` and `r.Cache.FindKeys(cacheId)`.

Reply:
Implemented the 2nd approach, for optimal availability while gradually updating the cache for possibly multiple hosts listed in the AuthConfig.